#### Урок 6. Сравнение долей. Построение доверительного интервала

1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.

In [24]:
g = 16
n = 256
x = 80

In [1]:
import numpy as np
from scipy import stats

In [27]:
z_tabl = stats.norm.ppf(0.975)
z_tabl

1.959963984540054

In [28]:
print(x-z_tabl*g/(n**0.5), x+z_tabl*g/(n**0.5))

78.04003601545995 81.95996398454005


Доверительный интервал: [78.04, 81.96]

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные:


6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1

Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

In [30]:
a =np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
a

array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [31]:
x_1 = np.mean(a)
x_1

6.590000000000001

In [32]:
D=np.var(a, ddof=1)
D

0.2032222222222223

In [33]:
t=stats.t.ppf(0.975, len(a)-1)
t

2.2621571627409915

In [35]:
print(x_1-t*np.sqrt(D/len(a)), x_1+t*np.sqrt(D/len(a)))

6.267515851415713 6.912484148584288


Т.о. интервал от 6.268 до 6.912, где в 95% случаях находится истинное математическое ожидание

3. Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170

Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175

Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

In [56]:
x =np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
y= np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])

In [60]:
x_x = np.mean(x)
x_x

166.2

In [61]:
x_y = np.mean(y)
x_y

168.1

In [62]:
D_x=np.var(x, ddof=1)
D_x

91.06666666666666

In [63]:
D_y=np.var(y, ddof=1)
D_y

60.10000000000001

In [64]:
delta = x_x-x_y
delta

-1.9000000000000057

In [65]:
D = (D_x + D_y)/2
D

75.58333333333334

In [66]:
Sdelta = (D/len(x)+D/len(y))**0.5
Sdelta

3.8880157750022915

In [67]:
t=stats.t.ppf(0.975,len(x)+ len(y)-2)
t

2.10092204024096

In [69]:
print(delta-t*Sdelta, delta+t*Sdelta)

-10.068418034506857 6.268418034506846


95% доверительный интервал для разности среднего роста родителей и детей составляет [-10.07, 6.27]